js/layer_type/symbol_layer_type.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+var Buffer = require('../data/buffer');
+
+var SYMBOL_ATTRIBUTES = [{
+    name: 'pos',
+    components: 2,
+    type: Buffer.AttributeType.SHORT,
+    value: function(x, y) {
+        return [x, y];
+    }
+}, {
+    name: 'offset',
+    components: 2,
+    type: Buffer.AttributeType.SHORT,
+    value: function(x, y, ox, oy) {
+        return [
+            Math.round(ox * 64), // use 1/64 pixels for placement
+            Math.round(oy * 64)
+        ];
+    }
+}, {
+    name: 'data1',
+    components: 4,
+    type: Buffer.AttributeType.UNSIGNED_BYTE,
+    value: function(x, y, ox, oy, tx, ty, minzoom, maxzoom, labelminzoom) {
+        return [
+            tx / 4, /* tex */
+            ty / 4, /* tex */
+            (labelminzoom || 0) * 10, /* labelminzoom */
+            0
+        ];
+    }
+}, {
+    name: 'data2',
+    components: 2,
+    type: Buffer.AttributeType.UNSIGNED_BYTE,
+    value: function(x, y, ox, oy, tx, ty, minzoom, maxzoom) {
+        return [
+            (minzoom || 0) * 10, /* minzoom */
+            Math.min(maxzoom || 25, 25) * 10 /* minzoom */
+        ];
+    }
```

Comment:
All the value functions seem a bit redundant to me. Can we refactor everything in a way so that we can move all this logic to corresponding buffer builders?